src/Plugins/PhpCodeSnifferJsonResultsParser/PhpCodeSnifferJsonResultsParser.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+/**
+ * Static Analysis Results Baseliner (sarb).
+ *
+ * (c) Dave Liddament
+ *
+ * For the full copyright and licence information please view the LICENSE file distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace DaveLiddament\StaticAnalysisResultsBaseliner\Plugins\PhpCodeSnifferJsonResultsParser;
+
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Common\FileName;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Common\InvalidPathException;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Common\LineNumber;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Common\Location;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Common\ProjectRoot;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Common\Type;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\File\InvalidFileFormatException;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\ResultsParser\AnalysisResult;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\ResultsParser\AnalysisResults;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\ResultsParser\Identifier;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\ResultsParser\ResultsParser;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Utils\ArrayParseException;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Utils\ArrayUtils;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Utils\FqcnRemover;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Utils\JsonParseException;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Utils\JsonUtils;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Utils\ParseAtLocationException;
+
+/**
+ * Handles PHPStan's JSON output.
+ *
+ * NOTE: SARB only deals with errors that are attached to a particular line in a file.
+ * PHPStan can report general errors (not specific to file). These are ignored by SARB.
+ */
+class PhpCodeSnifferJsonResultsParser implements ResultsParser
+{
+    private const LINE = 'line';
+    private const TYPE = 'type';
+    private const SOURCE = 'source';
+    private const FILES = 'files';
+    private const MESSAGES = 'messages';
+    private const MESSAGE = 'message';
+    private const ABSOLUTE_FILE_PATH = 'absoluteFilePath';
+    private const ERRORS = 'errors';
+    private const WARNINGS = 'warnings';
+
+    /**
+     * @var FqcnRemover
+     */
+    private $fqcnRemover;
+
+    /**
+     * PhpstanJsonResultsParser constructor.
+     */
+    public function __construct(FqcnRemover $fqcnRemover)
+    {
+        $this->fqcnRemover = $fqcnRemover;
+    }
+
```

Comment:
This is not needed for PHP Code Sniffer as it give a 'type', in this case the source rule, for each violation it finds.

For more info about this option [this doc] (https://github.com/DaveLiddament/sarb/blob/master/docs/ViolationTypeClassificationGuessing.md), so far PHPStan is the only tool which needs it.